test/SparqlAntlrParserTest.cpp

```diff
@@ -1114,3 +1115,20 @@ TEST(SparqlParser, SelectClause) {
   expectSelect("SELECT (5 - 10 as ?m) ?foo (10 as ?bar)",
                {Alias{"5-10", "?m"}, Variable{"?foo"}, Alias{"10", "?bar"}});
 }
+
+namespace {
+template <typename Exception = ParseException>
+void expectHavingConditionFails(const string& input) {
+  EXPECT_THROW(parseHavingCondition(input), Exception) << input;
+}
+}  // namespace
+
+TEST(SparqlParser, HavingCondition) {
+  auto expectHavingCondition = [](const string& input, SparqlFilter filter) {
+    expectCompleteParse(parseHavingCondition(input),
```

Comment:
Does `testing::Eq` consume its input? otherwise this is conceptually `const SparqlFilter&` as an argument and then no move, isn't it?